### ISSUE TITLE
Does not detect that switch case could be hit inside loop

### ISSUE THREAD

**Issue by ciaranmcnulty** [vimeo/psalm#5001]
Something wrong here with the flow analysis

https://psalm.dev/r/eef81dd77c

Removing the redundant conditional also removes the `TypeDoesNotContainType`

**Comment by muglug** [comment]
Reduced slightly to

```php
<?php

function analyse(): int {
    $state = 1;

    while (rand(0, 1)) {
        switch ($state) {
            case 3:
                break;
            case 2:
                if (rand(0, 1)) {
                    $state = 3;
                }
                break;
            default:
                $state = 2;
        }
    }
    
    return $state;
}
```
